doc/api/changes_dnf5.2.rst

```diff
@@ -0,0 +1,64 @@
+###############################################
+ Modifications to the public API in dnf 5.2.0.0
+###############################################
+
+This page lists the differences in the public API of `DNF5 5.2 <https://github.com/rpm-software-management/dnf5/releases/tag/5.2.0.0>`_ compared to the previous major version, DNF5 5.1.
+
+
+Wrapping struct attributes
+==========================
+
+Public attributes in struct objects have been wrapped in getter and setter methods.
+
+Before: ``bool ignore_case;`` Now: ``void set_ignore_case(bool); bool get_ignore_case() const;``
+
+* ``libdnf5::GoalJobSettings``
+* ``libdnf5::ResolveSpecSettings``
+* ``libdnf5::repo::RepoCacheRemoveStatistics``
+* ``libdnf5::rpm::Changelog``
+
+
+Dropping deprecated methods
+===========================
+
+Methods that were previously marked as deprecated have been removed or moved to the private section of the class.
+
+* ``libdnf5::create_file_logger(libdnf5::Base & base)`` (other variants still present)
+* ``libdnf5::base::Base::load_config_from_file()``
+* ``libdnf5::base::Base::with_config_file_path()``
+* ``libdnf5::repo::Repo::add_libsolv_testcase()``
+* ``libdnf5::repo::Repo::add_rpm_package()``
+* ``libdnf5::repo::Repo::download_metadata()``
+* ``libdnf5::repo::Repo::load()``
+* ``libdnf5::repo::Repo::load_extra_system_repo()``
+* ``libdnf5::repo::Repo::set_substitutions()``
+
+
+Dropping unused or redundant items
+==================================
+
+Methods and objects that were unused or redundant have been removed.
+
+* ``libdnf5::base::Base::get_comps()``
+* ``libdnf5::comps::Comps``
+* ``libdnf5::comps::EnvironmentSack``
+* ``libdnf5::comps::GroupSack``
+* ``libdnf5::repo::Repo::fresh()`` (use ``is_expired()`` instead)
+
+
+Other changes
+=============
+
+* ``libdnf5::advisory::AdvisoryQuery``
+
+  * ``filter_*`` methods have the default ``cmp_type`` ``IEXACT`` (was ``EQ`` before)
```

Comment:
What's the difference between `EQ` and `IEXACT`? Is it just that `IEXACT` is case-insensitive? What was the rationale for this change?

Reply:
Hmm, I still see `EQ` as the default in

https://github.com/rpm-software-management/dnf5/blob/1fe03c783b4bba7288eb4aa795718199102caa31/include/libdnf5/rpm/package_query.hpp#L83

and elsewhere. Am I missing something?

Reply:
Yes, `IEXACT` is just case insensitive. It's not all filter methods, just `filter_severity` and `filter_type`. The reason is that these two accept only specified values and it shouldn't matter if e.g. the type is "Bugfix" or "bugfix". See the commit: https://github.com/rpm-software-management/dnf5/commit/355aef18061bbee5599d70bd07da71d8db07c318

Reply:
Also, the `IEXACT` default is related to the mentioned `libdnf5::advisory::AdvisoryQuery` object. You are referencing the `libdnf5::rpm::PackageQuery` object which should be unchanged.

Reply:
But yeah, I've used `filter_*`, although just some methods were affected by this. Fixed now.